src/utils/createTheme.js

```diff
@@ -0,0 +1,160 @@
+                active: 8,
+                gradient: 5,
+                hover: 4,
+            },
+        },
+        midDark: {
+            yiqRange: [20, 118],
+            modifiers: {
+                active: -8,
+                gradient: -5,
+                hover: 4,
+            },
+        },
+        midTone: {
+            yiqRange: [118, 168],
+            modifiers: {
+                active: -7,
+                activeInverse: 9,
+                gradient: -7,
+                hover: 7,
+                hoverInverse: -4,
+            },
+        },
+        midLight: {
+            yiqRange: [168, 235],
+            modifiers: {
+                active: 15,
+                gradient: -10,
+                hover: 10,
+            },
+        },
+        vLight: {
+            yiqRange: [235, 256],
+            modifiers: {
+                active: -8,
+                gradient: -5,
+                hover: -4,
+            },
+        },
+    };
+
+    // Filter down the color map to the object that's in the proper YIQ range
+    const filteredColorMap = Object.keys(colorMap).reduce((obj, key) => {
+        if (
+            (colorKeyYiq >= colorMap[key].yiqRange[0] && colorKeyYiq < colorMap[key].yiqRange[1]) ||
+            (colorMap.lightnessThreshold && colorKeyLightness >= colorMap.lightnessThreshold)
+        ) {
+            obj = colorMap[key];
+            obj.range = key;
+        }
+
+        return obj;
+    }, {});
+
+    // Modify the primary colorKey with the associated modifiers from the filtered map
+    const modifiedColors = {};
+
+    for (const [key, value] of Object.entries(filteredColorMap.modifiers)) {
+        modifiedColors[key] = colorModifier(colorKeyObj, value);
+    }
+
+    // If the color is too extreme on either end of the spectrum we need to change our rules.
+    const exceedsLightThreshold = filteredColorMap.range === 'vLight';
+
+    // Light or dark isn't sufficient for determining how the secondary or accent colors should
+    // be calculated. In addition to that check, we will check the yiq value of the color to ensure
+    // the colorKey is not on the edges of the spectrum.
+
+    const hoverContrast = modifiedColors.hover.contrast(foregroundObj);
+    // If contrast has reached 21, we have hit the end of the spectrum and want to invert.
+    const hover =
+        hoverContrast >= MIN_CONTRAST + OFFSET_FACTOR && hoverContrast !== 21
+            ? modifiedColors.hover
+            : modifiedColors.hoverInverse || colorModifier(colorKeyObj, -filteredColorMap.modifiers.hover);
+
+    const activeContrast = modifiedColors.active.contrast(foregroundObj);
+    const active =
+        activeContrast >= MIN_CONTRAST + OFFSET_FACTOR && activeContrast !== 21
+            ? modifiedColors.active
+            : modifiedColors.activeInverse || colorModifier(colorKeyObj, -filteredColorMap.modifiers.active);
+
+    return {
+        primary: {
+            background: colorKey,
```

Comment:
maybe output `colorKeyObj.hex()` here?

Reply:
and for other colorKey references, so the file stays `hex` only instead of randomly having strings like `white`